recipes/LibriSpeech/ASR/seq2seq/train.py

```diff
@@ -292,7 +293,44 @@ def text_pipeline(wrd):
     sb.dataio.dataset.set_output_keys(
         datasets, ["id", "sig", "wrd", "tokens_bos", "tokens_eos", "tokens"],
     )
-    return train_data, valid_data, test_datasets
+    train_batch_sampler = None
+    valid_batch_sampler = None
+    if hparams["dynamic_batching"]:
+        from speechbrain.dataio.sampler import DynamicBatchSampler  # noqa
+        from speechbrain.dataio.dataloader import SaveableDataLoader  # noqa
+        from speechbrain.dataio.batch import PaddedBatch  # noqa
+
+        dynamic_hparams = hparams["dynamic_batch_sampler"]
+        hop_size = dynamic_hparams["feats_hop_size"]
+
+        num_buckets = dynamic_hparams["num_quantiles"]
+
+        train_batch_sampler = DynamicBatchSampler(
+            train_data,
+            dynamic_hparams["max_batch_len"],
+            num_buckets=num_buckets,
+            length_func=lambda x: x["duration"] * (1 / hop_size),
```

Comment:
I'm not sure to understand why we need to multiply by 1/hope_size. Isn't the duration enough for that? The sampler should operate on waveforms and it should be independent on the feature extraction process.

Reply:
we could avoid that. Personally it is easier for debugging and finding good values for `num_buckets` and `max_batch_len`.  At the first epoch you can look at the logs and intervals are easier to read in terms of feature frames.